lib/Cron/Manage.php

```diff
@@ -109,16 +110,32 @@ public function __construct(
 	 * @param $argument
 	 */
 	protected function run($argument) {
-		if (!$this->cronService->isRealCron()) {
+		if (!$this->cronService->isRunnable()) {
 			return;
 		}
 
+		try {
+			$this->cronService->lockCron(false);
+			$this->manage();
+			$this->cronService->unlockCron();
+		} catch (JobsTimeSlotException $e) {
+		}
+	}
+
+
+	/**
+	 * @throws JobsTimeSlotException
+	 */
+	private function manage() {
 		$generateLogs = $this->configService->getAppValueBool(ConfigService::GENERATE_LOGS);
 
 		// TODO: purge old restoring points.
 		$this->cronService->purgeRestoringPoints();
 		$this->cronService->purgeRemoteRestoringPoints();
 
+		// next steps are only available during night shift
+		$this->cronService->lockCron();
```

Comment:
Locked in `run(...)` above and locked here, is this possible?

Reply:
this one will check that we are running during night shift, and will update the time on the lock, in case previous process took a long time. This is usually the case of calling lockCron() after a process that might takes few minutes to run